extensions/Worker.Extensions.SignalRService/src/Hubs/ServiceHubContextInitializerOfT.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+
+namespace Microsoft.Azure.Functions.Worker.SignalRService
+{
+    internal class ServiceHubContextInitializer<THub, T> : ServiceHubContextInitializer<THub> where THub : ServerlessHub<T>
+        where T : class
+    {
+        private ServiceHubContext<T>? _serviceHubContext;
+
+        public ServiceHubContextInitializer(IConfiguration configuration, ILoggerFactory loggerFactory, HubContextProvider hubContextProvider, ServiceManagerOptionsSetup optionSetup) : base(configuration, loggerFactory, hubContextProvider, optionSetup)
+        {
+        }
+
+        public override async Task StartAsync(CancellationToken cancellationToken)
+        {
+            using var serviceManager = CreateServiceManager();
+            _serviceHubContext = await serviceManager.CreateHubContextAsync<T>(typeof(THub).Name, cancellationToken);
+            HubContextProvider.Add(typeof(THub), _serviceHubContext);
+        }
+
+        public override async Task StopAsync(CancellationToken cancellationToken)
+        {
+            if (_serviceHubContext != null)
+            {
+                await _serviceHubContext.DisposeAsync();
```

Comment:
I believe you should also make this type implement `IAsyncDisposable` and dispose `_serviceHubContext` if one is set.

There is an analyzer for this, but we may have it turned off. The gist is, if you ever hold disposables as fields, you must also be disposable yourself.

Reply:
`ServiceHubContext` is not `IAsyncDisposable`( The return type of `DisposeAsync` is Task instead of ValueTask due to historical reason), so I implement `IDisposable`. Also refactor a little bit.